frontend/public/components/deploy-image.tsx

```diff
@@ -0,0 +1,439 @@
+              targetPort: port.containerPort,
+              protocol: port.protocol,
+              // Use the same naming convention as CLI new-app.
+              name: `${port.containerPort}-${port.protocol}`.toLowerCase()
+            };
+          })
+        }
+      };
+
+      const createService = k8sCreate(ServiceModel, service)
+        .then(null, err => errorState(err));
+      promises.push(createService);
+    }
+
+    const imageStream = {
+      kind: 'ImageStream',
+      apiVersion: 'image.openshift.io/v1',
+      metadata: {
+        name,
+        namespace,
+        labels: {app: name}
+      },
+      spec: {
+        tags: [{
+          name: isi.tag,
+          annotations: _.assign({
+            'openshift.io/imported-from': isi.name
+          }, annotations),
+          from: {
+            kind: 'DockerImage',
+            name: isi.name
+          },
+          importPolicy: {}
+        }]
+      }
+    };
+
+    const createImageStream = k8sCreate(ImageStreamModel, imageStream)
+      .then(null, err => errorState(err));
+    promises.push(createImageStream);
+
+    // TODO: if one or more create fails, how to handle?
+    Promise.all(promises)
+      .then(() => {
+        this.setState({inProgress: false});
+        if (!this.state.error) {
+          history.push(resourcePathFromModel(DeploymentConfigModel, deploymentConfig.metadata.name, deploymentConfig.metadata.namespace));
+        }
+      });
+  };
+
+  render() {
+    const title = 'Deploy Image';
+    const { loading, isi, name, searchError } = this.state;
+
+    return <React.Fragment>
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <NavTitle title={title} />
+      <div className="co-m-pane__body">
+        <form className="" onSubmit={this.save}>
+          <div className="form-group co-deploy-image__namespace">
+            <label className="control-label co-required" htmlFor="dropdown-selectbox">Namespace</label>
+            <NsDropdown selectedKey={this.state.namespace} onChange={this.onNamespaceChange} id="dropdown-selectbox" />
+          </div>
+          <p>Deploy an existing image from an {/*image stream tag or */} image registry.</p>
+          <div className="form-group co-deploy-image__image-name">
+            <label htmlFor="imageName">Image Name</label>
+            <div className="input-group">
+              <input className="form-control"
+                type="search"
+                onChange={this.onImageNameChange}
+                onKeyDown={this.onKeyPress}
+                value={this.state.imageName}
+                id="imageName"
+                name="imageName" />
+              <span className="input-group-btn">
+                <button type="button" className="btn btn-default" onClick={this.search} disabled={this.state.imageName === ''}>
+                  <i className="fa fa-search" aria-hidden="true"></i>
+                  <span className="sr-only">Find</span>
```

Comment:
Probably better to add `aria-label` on the button

https://fontawesome.com/how-to-use/on-the-web/other-topics/accessibility